src/principles.ml

```diff
@@ -693,14 +693,14 @@ let subst_protos s gr =
     (equations_debug Pp.(fun () -> str"Fixed hint " ++ Printer.pr_econstr_env env sigma term);
      let sigma, _ = Typing.type_of env sigma term in
      let sigma = Evd.minimize_universes sigma in
-     Hints.IsConstr (Evarutil.nf_evar sigma term, Evd.universe_context_set sigma))
+     Hints.IsConstr (Evarutil.nf_evar sigma term, Some (Evd.universe_context_set sigma)))
```

Comment:
Is it always polymorphic?

Reply:
You always need to redeclare the constraints and freshen the term, so yes. Otherwise they are lost IIRC.